File: Python/DataStructures/Stack.py
class Node:
    def __init__(self, value, next):
        self.value = value
        self.next = next

class Stack:
    def __init__(self):
        self.head = None
        self.size = 0
    
    def isEmpty(self):
        return self.size <= 0

    def peek(self):
        if self.isEmpty():
            raise IndexError
        return self.head.value
    
    def push(self, value):
        if self.isEmpty():
            self.head = Node(value, None)
            self.size += 1
            return
        tmp = Node(value, self.head)
        self.head = tmp
        self.size += 1
    
    def pop(self):
        if self.isEmpty():
            raise IndexError
        tmp = self.head
        self.head = self.head.next
        self.size -= 1
        return tmp.value
    
    def poll(self):
        if self.isEmpty():
            return None
        tmp = self.head
        self.head = self.head.next
        self.size -= 1
        return tmp.value
    
    def __str__(self):
        res = "["
        tmp = self.head
        while tmp is not None:
            res += str(tmp.value)
            if tmp.next is not None:
                res += ", "
            tmp = tmp.next
        res += "]"
        return res

File: PHP/DataStructures/Stack.php
<?php

class Stack {
    // The top of the stack and the size.
    private $root;
    private $size;

    public function isEmpty() {
        return $this->size <= 0;
    }
    
    public function peek() {
        if($this->isEmpty()) {
            throw new Exception("Stack is empty!");
        }
        return $this->root->value;
    }

    public function push($value) {
        if($this->isEmpty()) {
            $this->root = new Node($value, null);
            $this->size++;
        }
        else {
            $tmp = new Node($value, $this->root);
            $this->root = $tmp;
            $this->size++;
        }
    }

    public function pop() {
        if($this->isEmpty()) {
            throw new Exception("Stack is empty!");
        }
        else {
            $tmp = $this->root;
            $this->root = $this->root->next;
            $this->size--;
            return $tmp->value;
        }
    }

    public function poll() {
        if($this->isEmpty()) {
            return null;
        }
        else {
            $tmp = $this->root;
            $this->root = $this->root->next;
            $this->size--;
            return $tmp->value;
        }
    }

    public function size() {
        return $this->size;
    }

    public function __toString() {
        $res = "{";
        $tmp = $this->head;
        while($tmp) {
            $res .= (string) $tmp->value;
            if($tmp->next) {
                $res .= ", ";
            }
            $tmp = $tmp->next;
        }
        return $res;
    }
}

// Supporting Node class for the stack
class Node {
    public $value;
    public $next;

    public function __constructor($value, $next) {
        $this->value = $value;
        $this->next = $next;
    }
}